pythonturtle/turtleprocess.py

```diff
@@ -51,11 +53,44 @@ def send_report(self):
 
     def run(self):
 
-        builtins.help = builtins.license = builtins.exit = \
-                                lambda *args, **kwargs: print('Not supported')
-
         self.turtle = Turtle()
 
+        def exit():
+            """
+            The built-in exit function terminates the interpreter hence
+            making the application unusable. We terminate the
+            application directly instead. NOTE: We use a proxy function
+            to hide the implementation details from the user.
+            """
+            evt = wx.CommandEvent(wx.ID_EXIT, wx.ID_EXIT)
+            wx.PostEvent(self.window, evt)
+
+        builtins.exit = exit
+
+        def help(object=None):
+            """
+            The built-in help function prints to the console and is
+            interactive there, thus blocking the UI. We simply display
+            the docstring of an object, when called with an argument,
+            or show the application help screen otherwise.
+            """
+            if object:
+                print(object.__doc__)
+            else:
+                self.window.show_help()
+
+        builtins.help = help
+
+        def license():
+            """
+            The built-in license function is interactive and blocks the
+            UI. We open the default web browser with the Python website
+            displaying the license instead.
+            """
+            webbrowser.open('https://docs.python.org/3/license.html')
```

Comment:
This is a compromise that does, of course, not work offline. We could try to find out whether the license is available as a file in a Python installation and is somehow accessible for printing out. Otherwise, I'd guess this is acceptable. Who will run this function anyway?

Reply:
This is definitely some very esoteric functionality. Your solution is above and beyond what we need.